src/EpicsClient/EpicsClientMonitor.cpp

```diff
@@ -191,94 +72,55 @@ int EpicsClientMonitor::emitWithoutCaching(
   return 0;
 }
 
-#define STRINGIFY2(x) #x
-#define STRINGIFY(x) STRINGIFY2(x)
-
-char const *channelStateName(epics::pvAccess::Channel::ConnectionState x) {
-#define DWTN1(N) DWTN2(N, STRINGIFY(N))
-#define DWTN2(N, S)                                                            \
-  if (x == epics::pvAccess::Channel::ConnectionState::N) {                     \
-    return S;                                                                  \
-  }
-  DWTN1(NEVER_CONNECTED);
-  DWTN1(CONNECTED);
-  DWTN1(DISCONNECTED);
-  DWTN1(DESTROYED);
-#undef DWTN1
-#undef DWTN2
-  return "[unknown]";
+std::string EpicsClientMonitor::getConnectionState() {
+  return toString(ConnectionState);
 }
 
-static std::string
-channelInfo(epics::pvAccess::Channel::shared_pointer const &Channel) {
-  std::ostringstream ss;
-  Channel->printInfo(ss);
-  return ss.str();
+void EpicsClientMonitor::handleConnectionStateChange(
+    ChannelConnectionState ConnectionState) {
+  LOG(Sev::Info, "EpicsClientMonitor::handleConnectionStateChange  {}",
+      toString(ConnectionState));
+  this->ConnectionState = ConnectionState;
+  if (ConnectionState == ChannelConnectionState::CONNECTED) {
+    Impl->monitoringStart();
+  } else {
+    Impl->monitoringStop();
+  }
+  if (ConnectionStatusProducer != nullptr) {
+    flatbuffers::FlatBufferBuilder Builder;
+    auto PVName = Builder.CreateString(Impl->channel_name);
```

Comment:
This case can be implemented in a different method. In this way `handleConnectionStateChange` only detects the connection state and delegates other methods to react accordingly (like you do in the case of `ChannelConnectionState::CONNECTED`)